python/GafferSceneUI/SceneInspector.py

```diff
@@ -987,6 +1002,18 @@ def __showHistory( self, target ) :
 		self.ancestor( GafferUI.Window ).addChildWindow( w, removeOnClose = True )
 		w.setVisible( True )
 
+	def __showShaderParameters( self, target ) :
+
+		w = _SectionWindow(
+			target.scene.node().getName() + " : " + self.__label().getText(),
+			_ShaderParameterSection( self.__label().getText(), self.__inspector, self.__diffCreator ),
+			target,
+			nodes = target.scene.ancestor( Gaffer.ScriptNode ).selection()
```

Comment:
This is where the "window closes when selection changes" problem is coming from. The _SectionWindow automatically closes itself when its nodeSet becomes empty, because it expects that it has been "pinned", in which case an empty node set indicates the node was deleted. This problem should go away when we replace `nodes` with `targets` in the `_SectionWindow` constructor, and derive the nodeSet from `targets`.

Reply:
Seems to be much better now, thanks!